src/setup-ganache.ts

```diff
@@ -20,25 +20,30 @@ export type AccountConfig = {
   secretKey: string;
 };
 
-export type Options = {
-  accountsConfig: AccountConfig[];
-  gasLimit: number;
-  gasPrice: number;
+export type NodeOptions = {
+  accounts: AccountConfig[];
   coverage: boolean;
+  gasPrice?: string;
+  gasLimit?: number;
+  allowUnlimitedContractSize?: boolean;
+  fork?: string;
+  unlocked_accounts?: string[];
 };
 
-export default async function(): Promise<string> {
+export default async function (): Promise<string> {
   const server = fork(path.join(__dirname, 'ganache-server'), [], {
     // Prevent the child process from also being started in inspect mode, which
     // would cause issues due to parent and child sharing the port.
     // See https://github.com/OpenZeppelin/openzeppelin-test-environment/pull/23
-    execArgv: process.execArgv.filter(opt => opt !== '--inspect'),
+    execArgv: process.execArgv.filter((opt) => opt !== '--inspect'),
   });
 
-  const options: Options = {
-    accountsConfig,
-    gasLimit: config.blockGasLimit,
-    gasPrice: config.gasPrice,
+  const gasPrice = typeof config.node.gasPrice === 'string' ? config.node.gasPrice : undefined;
+
+  const options: NodeOptions = {
+    ...config.node,
+    gasPrice,
```

Comment:
Isn't it possible to just forward `gasPrice` as it is? It's already either a string or undefined.

Reply:
No, because it wouldn't compile since `NodeOptions` has `gasPrice` type as a `string` only so we have to narrow type here to `string`.

Reply:
That's really interesting. The problem is basically that the "external" config type is more loosely typed (e.g. numbers can be written in many ways) but we want the internal config type to be more narrowly typed to what we will use in the rest of the library. We should refactor this into two separate types at some point (one type can extend the other).